crates/module-system/module-implementations/sov-evm/src/rpc/mod.rs

```diff
@@ -347,17 +347,17 @@ where
         debug!("EVM module JSON-RPC request to `eth_estimateGas`");
         let result = self.call(request, block_number, state)?;
         let gas_used = result.gas_used();
-        state
+        let gas_meter = state.try_as_basic_gas_meter().unwrap();
+        gas_meter
             .charge_linear_gas(
                 &<S as GasSpec>::gas_to_charge_per_evm_gas(),
                 gas_used as u32,
             )
             .unwrap();
-        let gas_meter = state.try_as_basic_gas_meter().unwrap();
         let total_gas_used =
             gas_meter.initial_gas.as_ref()[0] - gas_meter.remaining_gas.as_ref()[0];
-        const RELATIVE_MARGIN: u64 = 100_000;
-        let gas_used_with_margins = (total_gas_used * 3) / 2 + RELATIVE_MARGIN; // gas * 1.5 + 100_000
+        const ABSOLUTE_MARGIN: u64 = 100_000;
+        let gas_used_with_margins = (total_gas_used * 3) / 2 + ABSOLUTE_MARGIN; // gas * 1.5 + 100_000
```

Comment:
Let's use `checked_mul and checked_div`. The EVM module should never panic, even with extremely large values.
If the checked mul/div returns None then we should return an error. 

Also, calculating it as `(total_gas_used / 2) * 3 + ABSOLUTE_MARGIN`  reduces the likelihood of overflo. As the division is done before multiplication